ledger/participant-integration-api/src/main/scala/platform/apiserver/error/RejectionGenerators.scala

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.error
+
+import com.daml.error.BaseError
+import com.daml.ledger.participant.state
+import com.daml.lf.engine.Error.{Interpretation, Package, Preprocessing, Validation}
+import com.daml.lf.engine.{Error => LfError}
+import com.daml.lf.interpretation.{Error => LfInterpretationError}
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.platform.store.ErrorCause
+import io.grpc.Status.Code
+import io.grpc.StatusRuntimeException
+
+import scala.util.{Failure, Success, Try}
+
+object RejectionGenerators {
+
```

Comment:
I would inline this stuff in the ApiSubmissionService. The extraction only made sense in Canton.

Reply:
I have extracted it like this hoping that it could be easily leveraged by Canton as well when it will consume this API.